src/wallet/wallet2.cpp

```diff
@@ -1206,8 +1206,22 @@ void wallet2::refresh(uint64_t start_height, uint64_t & blocks_fetched, bool& re
   get_short_chain_history(short_chain_history);
   m_run.store(true, std::memory_order_relaxed);
   if (start_height > m_blockchain.size() || m_refresh_from_block_height > m_blockchain.size()) {
-    if (!start_height)
+    if (!start_height) {
+
+      std::string daemon_height_err = "";
+      uint64_t daemon_bc_height = this->get_daemon_blockchain_height(daemon_height_err);
```

Comment:
I agree. Better to just inform the user. 
It's kind of hard to follow your changes now because of a lot of whitespace changes.  
I think you should check your editor settings or commit/add using --ignore-whitespace to make the commits cleaner.